### PR TITLE
Sample app for executing GraphQL queries with Node.js

### PR DESCRIPTION
**Pull request by azlee**
Sample application in Node.js for executing GraphQL queries against [Square's GraphQL gateway](https://developer.squareup.com/docs/devtools/graphql) including merchant, catalog, customers, and orders queries.

### 1. Create a `.env` file

We've provided a file named `.env.example` that you can use as a starting point.  Its contents are as follows:

```bash
LOCATION_ID=<enter in a location id>
SQUARE_ACCESS_TOKEN=your-access-token
```

Copy this `.env.example` to a new file, named `.env`.  In the `.env` file, replace the placeholder text with valid values for your *LOCATION_ID* and *SQUARE_ACCESS_TOKEN*. We recommend using an access token from **sandbox** and not production.

### 2. Install dependencies
Use the Node package manager (`npm`) to install the dependencies you'll need for this program:
  
* `npm install`

### 3. Populate Test Data

The `bin/script/test-data.json` file contains fictitious data for:

* Catalog
* Customers
* Merchants
* Orders
  
To load this data into your **sandbox** environment, run the following command:

`npm run seed`

### 4. Run the sample program

After you've loaded the test data, you can use the sample program to run GraphQL queries.  Run the program as follows:

```bash
npm run start
```

The output should look similar to this:

```text
? Which GraphQL query would you like to execute? › - Use arrow-keys. Return to submit.
❯   catalog
    customers
    merchants
    orders
    payments
```

Choose `catalog` (the default).  

Next, you'll be prompted for your merchant ID:

```text
? Enter in your merchantId (required)
```

Type your merchant ID (see the *Prerequisites* section above), and then press Enter.

The first part of the output shows the GraphQL query to be executed:

```text
Running the query: 
  query CATALOG_SAMPLE_APP_QUERY(
    $after: Cursor,
    $merchantIds: [ID!]!
  ) {
    catalog {
      all(
        after: $after
        filter: {
          merchants: { equalToAnyOf: $merchantIds }
        }
      ) {
        pageInfo {
          ...PageInfoFragment
        }
        ...remaining output omitted...
  }
```

Next, it shows the results from executing the query:

```text
Page 1:
===================================================
{
  __typename: 'CatalogItem',
  id: 'BJL4CIX426EBHWJJORFEIR35',
  version: 1635975950679,
  isDeleted: false,
  updatedAt: '2021-11-03T21:45:50.679Z',
  absentAt: null,
  presentAt: null,
  presentAtAll: true,
  abbreviation: 'CR',
  availableElectronically: null,
  availableForPickup: null,
  availableOnline: null,
  category: { id: 'LIMNRASQPONR5CM4SWY3B5I6' },
  description: 'Light and flaky buttery croissant',
  image: null,
  options: null,
  labelColor: null,
  modifierListInfos: null,
  name: 'Croissant',
  productType: 'REGULAR',
  skipModifierScreen: null,
  taxes: [ { id: '7ANO66GMV5XZ2W2YEMBVQHLH', name: 'Sales tax' } ],
  variations: [
    { id: '3Y4YKX4AHV2GMYBUERBZEFKJ', name: 'Regular' },
    { id: 'M443PFRBMCQOFDZKVMRFAEP4', name: 'Chocolate Croissant' },
    { id: 'UP5PT4QOI7RICUMQ3DEDVAY7', name: 'Ham & Swiss Croissant' }
  ]
}
===================================================
   ...remaining output omitted...
```

### CODE REVIEW

**Comment by jessemartin** [comment]
Could we add `.env` to `.gitignore`, just to safeguard folks from committing credentials inadvertently?

File: connect-examples/v2/graphql-sample-app/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
+# Square and GraphQL - Code Example
```
Comment (jessemartin) [inline connect-examples/v2/graphql-sample-app/README.md:1]: Great instructions! I followed them and everything worked as expected without requiring any additional steps 👍

**Review by jessemartin — APPROVED** [review]

File: connect-examples/v2/graphql-sample-app/README.md (hunk 1 above)
Comment (azlee) [reply connect-examples/v2/graphql-sample-app/README.md:1]: nice! Thanks to @mcurtis-squareup for writing an awesome README!

**Review by jessemartin — APPROVED** [review]

**Review by jessemartin — APPROVED** [review]
LGTM, thanks for adding the .gitignore

**Pull request merged**